core/src/scene/stops.cpp

```diff
@@ -75,6 +75,61 @@ auto Stops::FontSize(const YAML::Node& _node) -> Stops {
     return stops;
 }
 
+auto Stops::Sizes(const YAML::Node& _node, const std::vector<Unit>& _units) -> Stops {
+    Stops stops;
+    if (!_node.IsSequence()) {
+        return stops;
+    }
+
+    for (auto& unit : _units) {
+        if (unit != Unit::pixel) {
+            LOGW("Size StyleParam can only take in pixel values.");
+            return stops;
+        }
+    }
+
+    float lastKey = 0;
+
+    for (const auto& frameNode : _node) {
+        if (!frameNode.IsSequence() || frameNode.size() != 2) { continue; }
+        float key = frameNode[0].as<float>();
+
+        if (lastKey > key) {
+            LOGW("Invalid stop order: key %f > %f\n", lastKey, key);
+            continue;
+        }
+        lastKey = key;
+
+        if (frameNode[1].IsScalar()) {
+            StyleParam::ValueUnitPair sizeValue;
+            sizeValue.unit = Unit::pixel;
+            size_t start = 0;
+
+            if (StyleParam::parseValueUnitPair(frameNode[1].Scalar(), start, sizeValue)) {
+                stops.frames.emplace_back(key, sizeValue.value);
```

Comment:
This is the point where the units should be tested against `_units` since it's the user parsed data, `sizeValue.unit` must be any `_units` value.